src/Voron/PageLocator.cs

```diff
@@ -77,9 +49,21 @@ public bool TryGetWritablePage(long pageNumber, out Page page)
 
             ref var node = ref Unsafe.Add(ref _cache[0], (int)bucket);
             page = node.Page;
-            return node.IsWritable && node.PageNumber == pageNumber && node.PageNumber != Invalid;
+            return node.Generation == _generation && node.IsWritable && node.PageNumber == pageNumber && node.PageNumber != Invalid;
         }
 
+        public void Renew()
+        {
+            _generation++;
+
+            // We are using 16 bits to store the generation, therefore if we are reusing the values after an overflow,
+            // we should reset the whole array. 
+            if (_generation == 0)
```

Comment:
This is wrong, this will only overflow every 2B items, but `Generation` is `ushort`

Reply:
That's correct. The likelihood of watching a bug there is negligible but it is still wrong.